cns/hnsclient/hnsclient_windows.go

```diff
@@ -685,3 +686,67 @@ func DeleteHostNCApipaEndpoint(
 
 	return nil
 }
+
+// DeleteHNSEndpointbyID deletes the HNS endpoint
+func DeleteHNSEndpointbyID(hnsEndpointID string) error {
+	var (
+		hcnEndpoint *hcn.HostComputeEndpoint
+		err         error
+	)
+
+	logger.Printf("Deleting hcn endpoint with id %v", hnsEndpointID)
+	hcnEndpoint, err = hcn.GetEndpointByID(hnsEndpointID)
+	if err != nil {
+		// If error is anything other than EndpointNotFoundError, return error.
+		// else log the error but don't return error because endpoint is already deleted.
+		if _, endpointNotFound := err.(hcn.EndpointNotFoundError); !endpointNotFound {
+			return fmt.Errorf("Failed to get hcn endpoint with id: %s due to err: %w", hnsEndpointID, err)
+		}
+
+		logger.Errorf("Delete called on the Endpoint which doesn't exist. Error:%v", err)
+		return nil
+	}
+
+	// Remove this endpoint from the namespace
+	if err = hcn.RemoveNamespaceEndpoint(hcnEndpoint.HostComputeNamespace, hcnEndpoint.Id); err != nil {
+		logger.Errorf("Failed to remove hcn endpoint %s from namespace %s due to err: %v", hcnEndpoint.Id, hcnEndpoint.HostComputeNamespace, err)
+	}
+
+	if err = hcnEndpoint.Delete(); err != nil {
+		return fmt.Errorf("Failed to delete endpoint: %+v. Error: %v", hnsEndpointID, err)
+	}
+
+	logger.Errorf("[Azure CNS] Successfully deleted endpoint: %+v", hnsEndpointID)
+
+	return nil
+}
+
+// GetHNSEndpointbyIP returns an HNSEndpoint with the corrsponding HNS Endpoint ID that matches an specific IP Address.
+func GetHNSEndpointbyIP(ipv4, ipv6 []net.IPNet, networkID string) (string, error) {
```

Comment:
why we pass ipv4, ipv6 separately.. can we detect keep it net.IPNet and based on v4/v6 we can search respective slices

Reply:
TH IPInfo struct has IPv4 and v6 separated in two different fields so we cannot pass a single parameter here.

Reply:
hmm ok.. can we pass ipinfo reference atleast?